### DISCUSSION TITLE
Stanford CS242: Programming Languages - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#494] [Announcements]
# Stanford CS242: Programming Languages - CS自学指南

CS自学指南

https://csdiy.wiki/%E7%A8%8B%E5%BA%8F%E8%AF%AD%E8%A8%80%E8%AE%BE%E8%AE%A1/CS242/

**Comment by Qi-Zhan** [comment]
这门课在2019年之后应该是回归了传统的 PL 风格教学，如 [2022](https://web.stanford.edu/class/cs242/materials.html)，供大家根据需要自行选择学习。

**Reply by riseOfCurse** [reply]
逆天课程，从leture 7开始Object Calculus就不对劲了，我正好再看那本 A Theory of Object，唉要是能有课程视频看就好了

**Reply by werner-balonius** [reply]
请问这节课怎么了吗，打算自学一下（p.s这节课咋没视频）

**Reply by riseOfCurse** [reply]
你好，逆天指的是这门课厉害，我没看过所有的lecture的pdf，我就看了一眼object calculus的那个lecture 7，里面内容和Luca写的《A Theory of Objects》的部分内容是一致的，其他内容不仅有lambda calculus以及continuation的概念，后面甚至连pi calculus都谈到了，我认为这门课对学习编程理论知识是有很大毗益的

**Reply by werner-balonius** [reply]
十分感谢！

**Reply by XFFFCCCC** [reply]
这个2022年份页面找不到lab唉

**Reply by Qi-Zhan** [reply]
我也没找到 2022 的 lab  和 homework :-(, 应该就是不公开的

**Reply by XFFFCCCC** [reply]
那只能做2019年的啦

**Reply by GowayLee** [reply]
我找到了2022的[homework仓库](https://github.com/danvk/Stanford-CS-242-Programming-Languages). 但是网上好像只有19年有详细的note, 22年的仓库以及实时当下的stanford官网都只有简略的课件, 感觉自学还是有一定障碍.

**Comment by lingkerio** [comment]
突然觉得看csdiy都是发现宝藏的过程，8.17号似乎增添了数门采用OCaml语言的课程。今晚突然发现更新了这门课程，在此之前我正在看北大的PL课程，在大概看完北大课程（或许双开）就来尝试一下这门课程。

**Reply by scatyf3** [reply]
是的，感觉养成习惯每天都要刷一刷csdiy

**Reply by Qi-Zhan** [reply]
<https://zhenjiang888.github.io/PL/2022/>

**Comment by featherwit001** [comment]
课程作业刚好都懂一点点，但是懂得不多。这个课程可以跟在CS3110之后，上手的难度曲线会平缓一些。

**Reply by xyuansui** [reply]
请问可以指导一下nfstar那个项目的环境怎么配吗

**Comment by Luv-Ray** [comment]
书面作业没有自测或者答案吗，对自己写的很没把握。。

**Reply by MTDickens** [reply]
同求，有没有写完的同学能把自己的书面作业答案发出来？

**Reply by heanyang1** [reply]
可以看看我写的答案：https://github.com/heanyang1/f19-assignments，虽然我也不保证一定对

我还把作业的代码移植到了 OCaml 5 和 Lean 4，用 Rust 重写了 assignment 4，想写编程作业的同学可以试一下

**Comment by MTDickens** [comment]
感觉可以先看北大的 PLT 课程，再看这个课程的 fa19 版本进阶、拓展。

北大课程链接：[讲义](https://zhenjiang888.github.io/PL/2022/lectures.html) [作业](https://zhenjiang888.github.io/PL/2022/assignments.html)

**Reply by MTDickens** [reply]
注：北大用的也是 OCaml。

**Reply by MTDickens** [reply]
正在找北大的朋友要录像，如果能要到的话，希望能够共享出来

**Reply by MTDickens** [reply]
录像看来是要不到了，因为北大的课程回放会在半年内删除。

**Reply by MTDickens** [reply]
不过讲义倒是写得不错。个人感觉看起来比 CS 242 更适合入门

**Reply by lingkerio** [reply]
实在可惜，北大的讲义还是精炼的

**Reply by Qi-Zhan** [reply]
确实，这个f19版本更像是一种课改的尝试，让理论的PL概念和真实世界更好的联系在一起。
如果是经典的PL学习，还是北大这种比较好。但很可惜的是，PL类的课程我基本都找不到视频

**Comment by zzzfreeuser** [comment]
求相关交流群
